packages/component/src/Transcript/FocusTrap.tsx

```diff
@@ -1,74 +1,115 @@
-
-import FocusRedirector from '../Utils/FocusRedirector';
 import tabbableElements from '../Utils/tabbableElements';
 import { useRefFrom } from 'use-ref-from';
 
-type FocusTrapProps = PropsWithChildren<{
+const FocusTrap = ({
+  children,
+  onFocus,
+  onLeave,
+  targetClassName
+}: Readonly<{
+  children: ReactNode;
   onFocus: () => void;
   onLeave: () => void;
-}>;
-
-const FocusTrap: FC<FocusTrapProps> = ({ children, onFocus, onLeave }) => {
+  targetClassName?: string | undefined;
+}>) => {
   const bodyRef = useRef<HTMLDivElement>();
-  const onLeaveRef = useRefFrom<() => void>(onLeave);
+  const lastFocused = useRef<HTMLElement>();
+  const onLeaveRef = useRefFrom(onLeave);
 
   const getTabbableElementsInBody = useCallback(
     () => tabbableElements(bodyRef.current).filter(element => element.getAttribute('aria-disabled') !== 'true'),
     [bodyRef]
   );
 
+  const focusOrTriggerLeave = useCallback(
+    (el: HTMLElement | undefined) => (el ? el.focus() : onLeaveRef.current?.()),
+    [onLeaveRef]
+  );
+
   const handleBodyKeyDown: KeyboardEventHandler = useCallback(
     event => {
       if (event.key === 'Escape') {
         event.preventDefault();
         event.stopPropagation();
 
         onLeaveRef.current?.();
+      } else if (event.key === 'Tab') {
+        const activeElement = document.activeElement as HTMLElement;
+        const focusables = getTabbableElementsInBody();
+        const focusedIndex = getTabbableElementsInBody().indexOf(activeElement);
+        if (event.shiftKey && focusedIndex === 0) {
+          event.preventDefault();
+          focusOrTriggerLeave(focusables.at(-1));
+        } else if (!event.shiftKey && focusedIndex === focusables.length - 1) {
+          event.preventDefault();
+          focusOrTriggerLeave(focusables.at(0));
+        }
       }
     },
-    [onLeaveRef]
+    [focusOrTriggerLeave, getTabbableElementsInBody, onLeaveRef]
   );
 
-  const handleFirstSentinelFocus: () => void = useCallback(() => {
-    const focusables = getTabbableElementsInBody();
-
-    const lastTabbableElement = focusables[focusables.length - 1];
+  const handleFocus = useCallback(
+    (event: FocusEvent<HTMLDivElement, Element>) => {
+      onFocus();
+      lastFocused.current = event.target;
+    },
+    [lastFocused, onFocus]
+  );
 
-    lastTabbableElement ? lastTabbableElement.focus() : onLeaveRef.current?.();
-  }, [getTabbableElementsInBody, onLeaveRef]);
+  const handleBlur = useCallback(
+    event => {
+      const { target } = event;
+      const focusables = getTabbableElementsInBody();
 
-  const handleLastSentinelFocus: () => void = useCallback(() => {
-    const [firstTabbableElement] = getTabbableElementsInBody();
+      // When blurred element became non-focusable, move to the first focusable element if available
+      // Otherwise trigger leave
+      if (target !== bodyRef.current && !focusables.includes(target)) {
```

Comment:
As we don't allow focusing container anymore, this can be simplified:
```suggestion
      if (!focusables.includes(target)) {
```